src/lib/adopted-probes.ts

```diff
@@ -0,0 +1,112 @@
+import type { Knex } from 'knex';
+import { scopedLogger } from './logger.js';
+import { client } from './sql/client.js';
+import { fetchSockets } from './ws/server.js';
+
+const logger = scopedLogger('adopted-probes');
+
+const TABLE_NAME = 'adopted_probes';
+
+type AdoptedProbe = {
+	ip: string;
+	uuid: string;
+	lastSyncDate: string;
+}
+
+export class AdoptedProbes {
+	private connectedIpToUuid: Map<string, string> = new Map();
+	private connectedUuidToIp: Map<string, string> = new Map();
+
+	constructor (
+		private readonly sql: Knex,
+		private readonly fetchWsSockets: typeof fetchSockets,
+	) {}
+
+	scheduleSync () {
+		setTimeout(() => {
+			this.syncDashboardData()
+				.finally(() => this.scheduleSync())
+				.catch(error => logger.error(error));
+		}, 5000);
```

Comment:
If we keep this running in multiple processes, let's bump the sync interval to 1 minute.